ibm/service/cis/data_source_ibm_cis_rulesets_entrypoint_version.go

```diff
@@ -0,0 +1,226 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package cis
+
+import (
+	"log"
+	"time"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	CISRulesetsEntryPointOutput = "rulesets"
+	CISRulesetPhase             = "ruleset_phase"
+	CISRulesetsPhaseListAll     = "list_all"
+	CISRulesetVersion           = "ruleset_version"
+)
+
+func DataSourceIBMCISRulesetsEntrypointVersions() *schema.Resource {
+	return &schema.Resource{
+		Read: dataIBMCISRulesetsEntrypointVersionsRead,
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+				ValidateFunc: validate.InvokeDataSourceValidator(
+					"ibm_cis_rulesets_versions",
+					"cis_id"),
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Optional:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			CISRulesetsId: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Id",
+			},
+			CISRulesetPhase: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Ruleset phase",
+			},
+			CISRulesetsPhaseListAll: {
```

Comment:
Check if optional fields conflict with each other, eg - `CISRulesetVersion`

Reply:
Check if this can be passed with value = true and then only the conflict appears for version